tests/Unit/EmogrifierTest.php

```diff
@@ -1725,6 +1725,85 @@ public function emogrifyIgnoresMediaQueryWithWhitespaceOnly()
         static::assertNotContains('@media screen', $result);
     }
 
+    /**
+     * @return mixed[][]
+     */
+    public function mediaTypesDataProvider()
+    {
+        return [
+            'disallowed type after disallowed type' => ['tv', 'speech', false],
+            'allowed type after disallowed type' => ['tv', 'all', true],
+            'disallowed type after allowed type' => ['screen', 'tv', false],
+            'allowed type after allowed type' => ['screen', 'all', true],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string $emptyRuleMediaType
+     * @param string $mediaType
+     * @param bool $isAllowedType
+     *
+     * @dataProvider mediaTypesDataProvider
+     */
+    public function emogrifyKeepsMediaRuleAfterEmptyMediaRuleIffAllowed($emptyRuleMediaType, $mediaType, $isAllowedType)
+    {
+        $this->subject->setHtml('<html><h1></h1></html>');
+        $mediaCss = '@media ' . $mediaType . ' { h1 { color: red; } }';
+        $this->subject->setCss('@media ' . $emptyRuleMediaType . ' {} ' . $mediaCss);
+
+        $result = $this->subject->emogrify();
+
+        if ($isAllowedType) {
```

Comment:
Please let's have two separate tests with separate data providers each for this.